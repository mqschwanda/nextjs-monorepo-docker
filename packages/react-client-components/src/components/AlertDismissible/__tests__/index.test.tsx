import {
  act, fireEvent, render, waitFor,
} from '@testing-library/react';
import { AlertDismissible } from '@';

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<AlertDismissible />', () => {
      it('renders', () => {
        const testId = AlertDismissible.name;
        const children = 'children...';

        const { getByTestId, asFragment } = render((
          <AlertDismissible
            className='test'
            variantBackgroundColor='info'
          >
            { children }
          </AlertDismissible>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();
        expect(element.classList.contains('hidden')).toBe(false);

        expect(asFragment()).toMatchSnapshot();
      });

      it('hides when dismiss button is clicked', () => {
        const testId = AlertDismissible.name;
        const children = 'children...';

        const { getByTestId } = render((
          <AlertDismissible
            className='test'
            variantBackgroundColor='info'
          >
            { children }
          </AlertDismissible>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();
        expect(element.classList.contains('hidden')).toBe(false);

        const [button] = element.getElementsByTagName('button');
        act(() => {
          fireEvent.click(button);
        });

        expect(element.classList.contains('hidden')).toBe(true);
      });

      it('hides when dismiss button is clicked and onClickDismiss prop returns true', async () => {
        const testId = AlertDismissible.name;
        const children = 'children...';
        const onClickDismiss = () => true;

        const { getByTestId } = render((
          <AlertDismissible
            className='test'
            onClickDismiss={onClickDismiss}
            variantBackgroundColor='info'
          >
            { children }
          </AlertDismissible>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();
        expect(element.classList.contains('hidden')).toBe(false);

        const [button] = element.getElementsByTagName('button');
        act(() => {
          fireEvent.click(button);
        });

        // TODO: cleanup tests and figure out why waitFor needed to be added
        await waitFor(() => {
          expect(element.classList.contains('hidden')).toBe(true);
        });
      });

      it('does not hide when dismiss button is clicked and onClickDismiss prop returns false', () => {
        const testId = AlertDismissible.name;
        const children = 'children...';
        const onClickDismiss = () => false;

        const { getByTestId } = render((
          <AlertDismissible
            className='test'
            onClickDismiss={onClickDismiss}
            variantBackgroundColor='info'
          >
            { children }
          </AlertDismissible>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();
        expect(element.classList.contains('hidden')).toBe(false);

        const [button] = element.getElementsByTagName('button');
        act(() => {
          fireEvent.click(button);
        });

        expect(element.classList.contains('hidden')).toBe(false);
      });
    });
  });
});
