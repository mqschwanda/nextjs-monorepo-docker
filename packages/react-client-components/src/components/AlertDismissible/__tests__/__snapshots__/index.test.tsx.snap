// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <AlertDismissible /> renders 1`] = `
<DocumentFragment>
  <div
    class="alert alert-info test"
    data-testid="AlertDismissible"
  >
    <div>
      children...
    </div>
    <button
      class="btn btn-circle btn-sm"
      type="button"
    >
      <svg
        class="stroke-current h-6 w-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
